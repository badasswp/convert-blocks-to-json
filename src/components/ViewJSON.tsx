import { __ } from '@wordpress/i18n';
import { select } from '@wordpress/data';
import { Button } from '@wordpress/components';

/**
 * View JSON.
 *
 * This Component returns the View JSON
 * label and button.
 *
 * @since 1.0.0
 *
 * @return {JSX.Element} View JSON.
 */
const ViewJSON = (): JSX.Element => {
	const postID = select( 'core/editor' ).getCurrentPostId();

	return (
		<>
			<p>{ __( 'View JSON', 'convert-blocks-to-json' ) }</p>
			<a
				href={ `${ cbtj.url }/wp-json/cbtj/v1/${ postID }` }
				target="_blank"
				rel="noreferrer"
			>
				<Button variant="primary" onClick={ () => {} }>
					{ __( 'View JSON', 'convert-blocks-to-json' ) }
				</Button>
			</a>
		</>
	);
};

export default ViewJSON;
