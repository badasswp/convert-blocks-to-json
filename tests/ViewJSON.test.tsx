import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';

import ViewJSON from '../src/components/ViewJSON';

jest.mock( '@wordpress/data', () => ( {
	select: jest.fn( ( storeName ) => {
		if ( storeName === 'core/editor' ) {
			return {
				getCurrentPostId: jest.fn( () => 1 ),
			};
		}
		return {};
	} ),
} ) );

jest.mock( '@wordpress/components', () => ( {
	Button: jest.fn( ( { children } ) => {
		return (
			<button type="button" className="components-button is-primary">
				{ children }
			</button>
		);
	} ),
} ) );

describe( 'ViewJSON', () => {
	beforeAll( () => {
		global.cbtj = { baseUrl: 'https://example.com' };
	} );

	it( 'renders the component with correct text and link', () => {
		const { container } = render( <ViewJSON /> );

		expect( container ).toMatchSnapshot();
	} );
} );
