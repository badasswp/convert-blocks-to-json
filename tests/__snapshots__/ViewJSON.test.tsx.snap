// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewJSON renders the component with correct text and link 1`] = `
<div>
  <p>
    View JSON
  </p>
  <a
    href="https://example.com/wp-json/cbtj/v1/1"
    rel="noreferrer"
    target="_blank"
  >
    <button
      class="components-button is-primary"
      type="button"
    >
      View JSON
    </button>
  </a>
</div>
`;
